internals/daemon/api_notices.go

```diff
@@ -182,5 +280,22 @@ func v1GetNotice(c *Command, r *http.Request, _ *UserState) Response {
 	if notice == nil {
 		return statusNotFound("cannot find notice with ID %q", noticeID)
 	}
+	if !noticeViewableByUser(notice, requestUID, daemonUID) {
```

Comment:
I'm wondering if for `GET /v1/notices` we should also loop through and check that they're all viewable by the user, just in case we have a logic bug in one of the places -- and probably `panic("internal error: ...")` if not. Or maybe our tests are enough :-) ... I haven't looked at those yet.

Reply:
This is a good question... I think the filtering is now so simple that this shouldn't be necessary (we only ever pass a `UserID` filter not matching `&requestUID` if the user is admin and explicitly asks for it).

If we really did want to make this check, I think we should skip the checking entirely for admins, since they may (in non-error cases as well) be looping through all notices, and that would be unnecessary and relatively slow. In that case, we'd probably want to split out the `isAdmin` check from within `noticeViewableByUser` so it doesn't do unnecessary work checking whether a user we know is non-admin is an admin. And probably `s/noticeViewableByUser/noticeViewableByNonAdmin/`.

I also don't like the idea of panicking there, at least outside of tests, which should I believe catch this anyway. We could easily handle it by not panicking and instead just excluding the notice. And I think if we really do want this check we should assume everything is fine and not do any memory allocations unless we catch an error, since they really shouldn't happen.

Reply:
So I'll leave it as-is for now, and if it's decided we would like a second pass check there, I can add it then.

Reply:
That's fine, given what you said, let's leave it as is.